src/pages/index.js

```diff
@@ -0,0 +1,253 @@
+import "./index.css";
+import {
+  initialCards,
+  avatarEditBtn,
+  profileEditBtn,
+  addNewCardBtn,
+  profileTitle,
+  profileSubtitle,
+  profileTitleInput,
+  profileSubtitleInput,
+  cardSelector,
+  validationSettings,
+  formValidators,
+} from "../utils/constants.js";
+import Card from "../components/Card.js";
+import FormValidator from "../components/FormValidator.js";
+import Section from "../components/Section.js";
+import PopupWithImage from "../components/PopupWithImage.js";
+import PopupWithForm from "../components/PopupWithForm.js";
+import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
+
+/* --------------------------------------------*/
+/* ------------------Validation----------------*/
+/* --------------------------------------------*/
+
+const enableValidation = (validationSettings) => {
+  const formList = document.querySelectorAll(validationSettings.formSelector);
+  formList.forEach((formElement) => {
+    const validator = new FormValidator(validationSettings, formElement);
+    const formName = formElement.getAttribute("name");
+
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+enableValidation(validationSettings);
+
+/* --------------------------------------------*/
+/* ------------------Event Handlers------------*/
+/* --------------------------------------------*/
+
+function handleCardImageClick(name, link) {
+  const data = { name, link };
+  popupWithImage.open(data);
+}
+
+function getCardElement(cardData) {
+  const card = new Card(
+    cardData,
+    cardSelector,
+    handleCardImageClick,
+    handleDeleteCard,
+    api.likeCard.bind(api),
+    api.deleteLike.bind(api),
+    confirmDeleteCard,
+    userId
+  );
+  return card.getView();
+}
+
+function renderCard(cardData) {
+  const cardElement = getCardElement(cardData);
+  const likeButton = cardElement.querySelector(".card__like-button");
+  if (cardData.isLiked) {
+    likeButton.classList.add("card__like-button_active");
+  } else {
+    likeButton.classList.remove("card__like-button_active");
```

Comment:
Please, delete it from `renderCard`

Now only the `Card` class knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle


It should be done **inside** the `Card` class where you've already found the like button, and you have `cardData` there as well

Here will be only this:

```js

function renderCard(cardData) {
  const cardElement = getCardElement(cardData);
  section.addItem(cardElement);
}


```